drivers/hmis/spec/requests/hmis/delete_enrollment_spec.rb

```diff
@@ -58,6 +64,16 @@ def perform_mutation(enrollment)
     expect(e2.reload).not_to be_deleted
   end
 
+  it 'should allow deleting non-WIP enrollments if no intake assessment exists' do
+    add_permissions(access_control, :can_edit_enrollments, :can_delete_enrollments)
+    enrollment, errors = perform_mutation(e3)
+    expect(enrollment).to be_present
+    expect(errors).to be_empty
+    expect(e1.reload).not_to be_deleted
+    expect(e2.reload).not_to be_deleted
+    expect(e3.reload).to be_deleted
+  end
```

Comment:
Thanks for including tests 🙌  

I might be missing something but do we need to test e1 and e2? It seems unnecessary to me - @gigxz what do you think? If we really want to, maybe we could just have one dedicated test for this: "Enrollment deletion only deletes the selected enrollment".

Also I think we are pretty inconsistent with how we structure tests. This is subjective but I'd suggest using something like the following pattern:
* use an enclosing context to to help clarify what variables like e3 are for
* use an `expect` block to concisely check for expected changes

```suggestion
  context 'a completed enrollment without an intake assessment' do
    let(:c3) { create :hmis_hud_client, data_source: ds1, user: u1 }
    let!(:e3) { create :hmis_hud_enrollment, data_source: ds1, project: p1, client: c3, relationship_to_ho_h: 1 }
    it 'should be deleted successfully if authorized' do
      add_permissions(access_control, :can_edit_enrollments, :can_delete_enrollments)
      expect do
        enrollment, errors = perform_mutation(e3)
        expect(enrollment).to be_present
        expect(errors).to be_empty
      end.to change(Hmis::Hud::Enrollment.where(id: e3.id), :count).by(-1)
    end
    it 'should throw error if unauthorized' do
      expect_gql_error post_graphql(input: { id: e3.id }) { mutation }
    end
  end
```

Reply:
Thanks @ttoomey for your feedback!
> do we need to test e1 and e2?

No probably not! I think I just did this because I was following the example from line 63-64, but I agree that it would be clearer if the tests all have a single purpose and separate contexts. I'll open a new PR since this one is already merged.

Reply:
https://github.com/greenriver/hmis-warehouse/pull/3945